app/routers/categories.py

```diff
@@ -68,11 +75,24 @@ def validate_request_params(query_params: ImmutableMultiDict) -> bool:
     Intersection must contain at least user_id.
     Union must not contain fields other than user_id, name, color.
     """
+    is_valid_color = True
     all_fields = set(CategoryModel.schema()["required"])
     request_params = set(query_params)
     union_set = request_params.union(all_fields)
     intersection_set = request_params.intersection(all_fields)
-    return union_set == all_fields and "user_id" in intersection_set
+    if "color" in intersection_set:
+        is_valid_color = validate_color_format(query_params["color"])
+    return union_set == all_fields and "user_id" in intersection_set and (
+        is_valid_color)
+
+
+def validate_color_format(color: str) -> bool:
+    """
+    Validate color is from hex format (without `#`)
+    """
+    if color:
+        return re.match(HEX_COLOR_FORMAT, color) is not None
```

Comment:
Awesome. I'd maybe suggest to use `re.fullmatch` to make it even easier to read.